Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.h

```diff
@@ -455,74 +464,122 @@ namespace Terrain
     {
         AZ_PROFILE_FUNCTION(Terrain);
 
-        // ToDo: Determine the number of jobs to split the work into based on:
-        // 1. The number of available worker threads.
-        // 2. The desired number of jobs as passed in.
-        // 3. The size of the area being processed.
-        //
-        // Note: We are currently restricting the number of worker threads to one
-        // because splitting the work over multiple threads causes contention when
-        // locking various mutexes, resulting in slower overall wall time for async
-        // requests split over multiple threads vs one where all the work is done on
-        // a single thread. The latter is still preferable over a regular synchronous
-        // call because it is just as quick and prevents the main thread from blocking.
-        // Once the mutex contention issues have been addressed, we should come up with
-        // an algorithm to break up 'inRegion' into sub-regions (or lists of positions?)
-        // so that async calls automatically split the work between available job manager
-        // worker threads, unless the ProcessAsyncParams specifiy a desired number of jobs.
-        const int32_t numWorkerThreads = m_terrainJobManager->GetNumWorkerThreads();
-        const int32_t numJobsDesired = params ? params->m_desiredNumberOfJobs : ProcessAsyncParams::NumJobsDefault;
-        int32_t numJobs = (numJobsDesired > 0) ? AZStd::min(numWorkerThreads, numJobsDesired) : numWorkerThreads;
-        if (numJobs != 1)
+        const auto [numSamplesX, numSamplesY] = GetNumSamplesFromRegion(inRegion, stepSize, sampler);
+        const int64_t numPositionsToProcess = numSamplesX * numSamplesY;
+
+        if (numPositionsToProcess == 0)
         {
-            // Temp until we figure out how to break up the region.
-            AZ_Warning("TerrainSystem", false, "We don't yet support breaking up regions.");
-            numJobs = 1;
+            AZ_Warning("TerrainSystem", false, "No positions to process.");
+            return nullptr;
         }
 
+        // Determine the *ideal* maximum number of jobs to split the work into based on:
+        // 1. The maximum number of jobs desired and available.
+        // 2. The minimum number of positions to process per job.
+        // 3. The total number of positions being processed.
+        const int32_t numWorkerThreads = m_terrainJobManager->GetNumWorkerThreads();
+        const int32_t numJobsDesired = params ? params->m_desiredNumberOfJobs : ProcessAsyncParams::NumJobsDefault;
+        const int32_t numJobsMax = (numJobsDesired > 0) ? AZStd::min(numWorkerThreads, numJobsDesired) : numWorkerThreads;
+        const int32_t minPositionsPerJob =
+            params && (params->m_minPositionsPerJob > 0) ? params->m_minPositionsPerJob : ProcessAsyncParams::MinPositionsPerJobDefault;
+        int32_t numJobs = AZStd::clamp(aznumeric_cast<int32_t>(numPositionsToProcess / minPositionsPerJob), 1, numJobsMax);
```

Comment:
Mixing this request with the other one, I've moved the first 3 lines of the above into a shared helper method, left the minPositionsPerJob duplicated because there's nothing really special about it, and moved the ProcessRegion version of the numJobs clamp down into SubdivideRegionForJobs() so that the clamp is being enforced and used in context of the overall calculation. The clamp is duplicated in the ProcessList method, but it's because that's the entire calculation used for subdividing the list into jobs.